test/e2e/proxy_test.go

```diff
@@ -0,0 +1,479 @@
+			},
+			Spec: appsv1.DeploymentSpec{
+				Replicas: func() *int32 { i := int32(1); return &i }(),
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{"app": "squid"},
+				},
+				Template: corev1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{"app": "squid"},
+					},
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name:  "squid",
+								Image: "ubuntu/squid:edge",
+								Ports: []corev1.ContainerPort{
+									{
+										ContainerPort: httpsPort,
+										Name:          "squid-https",
+										Protocol:      corev1.ProtocolTCP,
+									},
+									{
+										ContainerPort: httpPort,
+										Name:          "squid",
+										Protocol:      corev1.ProtocolTCP,
+									},
+								},
+								VolumeMounts: []corev1.VolumeMount{
+									{
+										Name:      "squid-config-volume",
+										MountPath: "/etc/squid/squid.conf",
+										SubPath:   "squid.conf",
+									},
+									{
+										Name:      "squid-data",
+										MountPath: "/var/spool/squid",
+									},
+									{
+										Name:      "squid-ssl",
+										MountPath: "/etc/squid/ssl_ca",
+										ReadOnly:  true,
+									},
+									{
+										Name:      "squid-logs",
+										MountPath: "/var/log/squid",
+									},
+								},
+							},
+						},
+						Volumes: []corev1.Volume{
+							{
+								Name: "squid-config-volume",
+								VolumeSource: corev1.VolumeSource{
+									ConfigMap: &corev1.ConfigMapVolumeSource{
+										LocalObjectReference: corev1.LocalObjectReference{
+											Name: squidConfigName,
+										},
+										Items: []corev1.KeyToPath{
+											{
+												Key:  "squid",
+												Path: "squid.conf",
+											},
+										},
+									},
+								},
+							},
+							{
+								Name: "squid-data",
+								VolumeSource: corev1.VolumeSource{
+									PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+										ClaimName: PVCName,
+									},
+								},
+							},
+							{
+								Name: "squid-ssl",
+								VolumeSource: corev1.VolumeSource{
+									Secret: &corev1.SecretVolumeSource{
+										SecretName: "squid-service-tls",
+										Items: []corev1.KeyToPath{
+											{
+												Key:  "tls.crt",
+												Path: "tls.crt",
+											},
+											{
+												Key:  "tls.key",
+												Path: "tls.key",
+											},
+										},
+									},
+								},
+							},
+							{
+								Name: "squid-logs",
+								VolumeSource: corev1.VolumeSource{
+									EmptyDir: &corev1.EmptyDirVolumeSource{},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		err = client.Create(deployment)
+		Expect(err).NotTo(HaveOccurred())
+		err = client.WaitForDeploymentRollout(deployment)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("copy contents of openshift-service-ca.crt to proxy-ca")
+		serviceCAConfigMap := &corev1.ConfigMap{
```

Comment:
we need a jira ticket to remind us make the CA keyname costomizable in the configmap.

Reply:
yes, working on creating jira tickets now

Reply:
[OLS-1998](https://issues.redhat.com/browse/OLS-1998)